test/pool/FeesEscrow.test.js

```diff
@@ -0,0 +1,117 @@
+const { contracts, contractSettings } = require('../../deployments/settings');
+const {setupOracleAccounts, impersonateAccount} = require("../utils");
+const {upgradeContracts} = require("../../deployments");
+const {send, ether, BN} = require("@openzeppelin/test-helpers");
+const {ethers} = require("hardhat");
+const {expect} = require("chai");
+
+const Pool = artifacts.require('Pool');
+const StakedEthToken = artifacts.require('StakedEthToken');
+const RewardEthToken = artifacts.require('RewardEthToken');
+const PoolValidators = artifacts.require('PoolValidators');
+const Oracles = artifacts.require('Oracles');
+
+let pool;
+let feesEscrow;
+let stakedEthToken;
+let rewardEthToken;
+let validators;
+let oracles;
+let oracleAccounts;
+const admin = contractSettings.admin;
+
+async function upgradeRewardEthToken() {
+  const signer = await ethers.provider.getSigner(contractSettings.admin);
+  const RewardEthToken = await ethers.getContractFactory(
+    'RewardEthToken',
+    signer
+  );
+  let rewardEthToken = await RewardEthToken.attach(contracts.rewardEthToken);
+
+  // pause
+  if (!(await rewardEthToken.paused())) {
+    await rewardEthToken.pause();
+  }
+
+  // upgrade RewardEthToken to new implementation
+  const proxy = await upgrades.upgradeProxy(
+    contracts.rewardEthToken,
+    RewardEthToken,
+    {
+      unsafeAllowRenames: true,
+    }
+  );
+  await proxy.deployed();
+
+  const [owner] = await ethers.getSigners();
+
+  return rewardEthToken.unpause();
+}
+
+contract('FeesEscrow', (accounts) => {
```

Comment:
Please add test to verify that transferToPool can be called only by the RewardEthToken